snuba/datasets/schemas/join.py

```diff
@@ -115,6 +122,11 @@ def get_columns(self) -> QualifiedColumnSet:
         column_sets = {alias: table.get_columns() for alias, table in tables.items()}
         return QualifiedColumnSet(column_sets)
 
+    def get_mandatory_conditions(self) -> Sequence[Condition]:
```

Comment:
Still considering whether it would make sense to have this method only in classes that extend TableSource instead of any RelationalSource. Having it at the top makes things a little cleaner when applying these conditions.